Assets/LeapMotion/Scripts/DetectionUtilities/FingerDirectionDetector.cs

```diff
@@ -0,0 +1,122 @@
+﻿using UnityEngine;
+using System.Collections;
+using Leap;
+
+namespace Leap.Unity {
+
+  public class FingerDirectionDetector : Detector {
+    [Tooltip("The interval in seconds at which to check this detector's conditions.")]
+    public float Period = .1f; //seconds
+    [Tooltip("The hand model to watch. Set automatically if detector is on a hand.")]
+    public IHandModel handModel = null;  
+    public Finger.FingerType FingerName = Finger.FingerType.TYPE_INDEX;
+    [Tooltip("The target direction.")]
+    public Vector3 PointingDirection = Vector3.forward;
+    [Tooltip("How to treat the target direction.")]
+    public PointingType PointingType = PointingType.RelativeToHorizon;
+    [Tooltip("A target object(optional). Use PointingType.AtTarget")]
+    public Transform TargetObject = null;
+    [Tooltip("The angle in degrees from the target direction at which to turn on.")]
+    [Range(0, 360)]
+    public float OnAngle = 15f; //degrees
+    [Tooltip("The angle in degrees from the target direction at which to turn off.")]
+    [Range(0, 360)]
+    public float OffAngle = 25f; //degrees
+
+    private void OnValidate(){
+      if( OffAngle < OnAngle){
+        OffAngle = OnAngle;
+      }
+    }
+
+    private void Start () {
+      if(handModel == null){
+        handModel = gameObject.GetComponentInParent<IHandModel>();
+      }
+    }
+
+    private void OnEnable () {
+      StartCoroutine(fingerPointingWatcher());
+    }
+  
+    private void OnDisable () {
+      StopCoroutine(fingerPointingWatcher());
```

Comment:
Odd, I thought I had switched them all over to saving a reference to the IEnumerator... Well, they do now.